wait/host.go

```diff
@@ -584,51 +585,77 @@ func (a *HostAwaitility) WaitForTemplateUpdateRequests(namespace string, count i
 }
 
 // NotificationWaitCriterion checks if a Notification meets the given condition
-type NotificationWaitCriterion func(a *HostAwaitility, mur *toolchainv1alpha1.Notification) bool
+type NotificationWaitCriterion func(a *HostAwaitility, mur toolchainv1alpha1.Notification) bool
 
-// WaitForNotification waits until there is a Notification available with the given name and the optional conditions
-func (a *HostAwaitility) WaitForNotification(name string, criteria ...NotificationWaitCriterion) (*toolchainv1alpha1.Notification, error) {
-	var notification *toolchainv1alpha1.Notification
+// WaitForNotifications waits until there is an expected number of Notifications available for the provided user and with the notification type and which match the conditions (if provided).
+func (a *HostAwaitility) WaitForNotifications(username, notificationType string, numberOfNotifications int, criteria ...NotificationWaitCriterion) ([]*toolchainv1alpha1.Notification, error) {
+	var notifications []*toolchainv1alpha1.Notification
 	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
-		obj := &toolchainv1alpha1.Notification{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj); err != nil {
-			if errors.IsNotFound(err) {
-				a.T.Logf("waiting for availability of notification '%s'", name)
-				return false, nil
-			}
+		labels := map[string]string{toolchainv1alpha1.NotificationUserNameLabelKey: username, toolchainv1alpha1.NotificationTypeLabelKey: notificationType}
+		opts := client.MatchingLabels(labels)
+		notificationList := &toolchainv1alpha1.NotificationList{}
+		if err :=  a.Client.List(context.TODO(), notificationList, opts);  err != nil {
 			return false, err
 		}
-		for _, match := range criteria {
-			if !match(a, obj) {
-				return false, nil
+
+		actualNotificationCount := len(notificationList.Items)
+		if numberOfNotifications < actualNotificationCount{
```

Comment:
instead of having 2 blocks to compare the actual number of notifications vs the expected one, you could probably just do:
```
		if numberOfNotifications != actualNotificationCount {
			a.T.Logf("expected '%d' notifications, but found '%d' notifications", numberOfNotifications, actualNotificationCount)
			return false, errs.New(fmt.Sprintf("expected %d notifications, but found %d notifications", numberOfNotifications, actualNotificationCount))
		}
```

Reply:
@xcoulon Hmm, I'm not sure this would work? I think we would need to know whether the expected notifications is less than the number we actually got in the list - in this case, we aren't waiting for more notifications. I was thinking this would probably be an error in our expectation or the test failing for some reason. On the other hand, if we have less notifications than what we expect, we should be waiting to see if more appear I guess? What do you think?

Reply:
Fixed here: b32b8a788930c4cfc225b932d8c5396daed3b270 to wait until we get an exact match or a timeout will occur